packages/components/tabs/src/tab-bar.vue

```diff
@@ -23,79 +16,67 @@ import { tabBar } from './tab-bar'
-    const barStyle = ref()
+const instance = getCurrentInstance()!
+const rootTabs = inject(tabsRootContextKey)
+if (!rootTabs) throwError(COMPONENT_NAME, 'must use with ElTabs')
 
-    const getBarStyle = (): CSSProperties => {
-      let offset = 0
-      let tabSize = 0
+const bar$ = ref<HTMLDivElement>()
+const barStyle = ref()
 
-      const sizeName = ['top', 'bottom'].includes(rootTabs.props.tabPosition)
-        ? 'width'
-        : 'height'
-      const sizeDir = sizeName === 'width' ? 'x' : 'y'
+const getBarStyle = (): CSSProperties => {
+  let offset = 0
+  let tabSize = 0
 
-      props.tabs.every((tab) => {
-        const $el = instance.parent?.refs?.[
-          `tab-${tab.paneName}`
-        ] as HTMLElement
-        if (!$el) return false
+  const sizeName = ['top', 'bottom'].includes(rootTabs.props.tabPosition)
+    ? 'width'
+    : 'height'
+  const sizeDir = sizeName === 'width' ? 'x' : 'y'
 
-        if (!tab.active) {
-          return true
-        }
+  props.tabs.every((tab) => {
+    const $el = instance.parent?.refs?.[`tab-${tab.paneName}`] as HTMLElement
+    if (!$el) return false
 
-        tabSize = $el[`client${capitalize(sizeName)}`]
-        const position = sizeDir === 'x' ? 'left' : 'top'
-        offset =
-          $el.getBoundingClientRect()[position] -
-          ($el.parentElement?.getBoundingClientRect()[position] ?? 0)
-        const tabStyles = window.getComputedStyle($el)
+    if (!tab.active) {
+      return true
+    }
 
-        if (sizeName === 'width') {
-          if (props.tabs.length > 1) {
-            tabSize -=
-              parseFloat(tabStyles.paddingLeft) +
-              parseFloat(tabStyles.paddingRight)
-          }
-          offset += parseFloat(tabStyles.paddingLeft)
-        }
-        return false
-      })
+    tabSize = $el[`client${capitalize(sizeName)}`]
+    const position = sizeDir === 'x' ? 'left' : 'top'
+    offset =
+      $el.getBoundingClientRect()[position] -
+      ($el.parentElement?.getBoundingClientRect()[position] ?? 0)
+    const tabStyles = window.getComputedStyle($el)
 
-      return {
-        [sizeName]: `${tabSize}px`,
-        transform: `translate${capitalize(sizeDir)}(${offset}px)`,
+    if (sizeName === 'width') {
+      if (props.tabs.length > 1) {
+        tabSize -=
+          parseFloat(tabStyles.paddingLeft) + parseFloat(tabStyles.paddingRight)
       }
+      offset += parseFloat(tabStyles.paddingLeft)
     }
+    return false
+  })
 
-    const update = () => (barStyle.value = getBarStyle())
+  return {
+    [sizeName]: `${tabSize}px`,
+    transform: `translate${capitalize(sizeDir)}(${offset}px)`,
+  }
+}
 
-    watch(
-      () => props.tabs,
-      async () => {
-        await nextTick()
-        update()
-      },
-      { immediate: true }
-    )
-    useResizeObserver(bar$, () => update())
+const update = () => (barStyle.value = getBarStyle())
 
-    return {
-      bar$,
-      rootTabs,
-      barStyle,
-      update,
-    }
```

Comment:
I think we should use `defineExpose()` to expose these API

Reply:
Yes, you're right.